crates/flox/src/commands/general.rs

```diff
@@ -137,40 +138,28 @@ impl ConfigArgs {
             key: impl AsRef<str>,
             value: Option<V>,
         ) -> Result<()> {
-            let key_path = Key::parse(key.as_ref()).context("Could not parse key")?;
-
-            let config_file_path = config_dir.join("flox.toml");
-            let config_file_contents = match fs::read_to_string(&config_file_path).await {
-                Ok(s) => Ok(Some(s)),
-                Err(e) if e.kind() == io::ErrorKind::NotFound => {
-                    warn!("No existing user config file found");
-                    Ok(None)
-                },
-                Err(e) => Err(e).context("Could read current config file.\nPlease verify the format or reset using `flox config --reset`"),
-            }?;
-            let config_file_contents = Config::write_to(config_file_contents, &key_path, value)
-                .context("Could not write config value")?;
-
-            write_config(temp_dir, config_dir, config_file_contents).await?;
-            Ok(())
-        }
+            let query = Key::parse(key.as_ref()).context("Could not parse key")?;
 
-        async fn write_config(
-            temp_dir: &Path,
-            config_dir: &Path,
-            content: impl AsRef<[u8]>,
-        ) -> Result<()> {
-            let tempfile = tempfile::Builder::new().tempfile_in(temp_dir)?;
-            fs::write(&tempfile, content)
-                .await
-                .context("Could not write config file")?;
-            tempfile.persist(config_dir.join("flox.toml"))?;
+            let config_file_path = config_dir.join(FLOX_CONFIG_FILE);
+
+            match Config::write_to_in(config_file_path, temp_dir, &query, value) {
+                err @ Err(ReadWriteError::ReadConfig(_)) => err.context("Could read current config file.\nPlease verify the format or reset using `flox config --reset`")?,
+                err@ Err(_) => err?,
```

Comment:
```suggestion
                err @ Err(_) => err?,
```

strangely rustfmt didn't care about that